src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerConversionExtensions.cs

```diff
@@ -43,13 +43,46 @@ public static class JaegerConversionExtensions
         private const long TicksPerMicrosecond = TimeSpan.TicksPerMillisecond / 1000;
         private const long UnixEpochMicroseconds = UnixEpochTicks / TicksPerMicrosecond; // 62,135,596,800,000,000
 
+        private static readonly Dictionary<string, int> PeerServiceKeyResolutionDictionary = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase)
+        {
+            ["peer.service"] = 0, // peer.service primary.
+            ["net.peer.name"] = 1, // peer.service first alternative.
+            ["peer.hostname"] = 2, // peer.service second alternative.
+            ["peer.address"] = 2, // peer.service second alternative.
+            ["http.host"] = 3, // peer.service for Http.
+            ["db.instance"] = 4, // peer.service for Redis.
+        };
+
         public static JaegerSpan ToJaegerSpan(this SpanData span)
         {
-            List<JaegerTag> jaegerTags = null;
+            var jaegerTags = new List<JaegerTag>();
+
+            Tuple<string, int> peerService = null;
+            foreach (var label in span.Attributes)
+            {
+                var tag = label.ToJaegerTag();
+
+                if (tag.VStr != null
+                    && PeerServiceKeyResolutionDictionary.TryGetValue(label.Key, out int priority)
+                    && (peerService == null || peerService.Item2 > priority))
```

Comment:
I think this should be `<` as the highest priority is in ascending order from 0 - we're looking for the lowest priority.

Reply:
I switched them around so it reads better.